script.R

```diff
@@ -10,6 +10,53 @@ if_error <- function(x, y = NULL) {
 `%||%` <- function(x, y) {
   if (!length(x) || is.null(x)) y else x
 }
+check_if_pkg_available <- function(pkg, ver = NULL) {
```

Comment:
Two possible other implementation of this, without that much repetition

```r
check_if_pkg_available <- function(pkg, ver = NULL) {
  length(
    available.packages(
      filters = list(
        add = TRUE,
        function(db) {
          if (is.null(ver)) {
            db[db[, "Package"] == pkg, ]
          } else {
            db[db[, "Package"] == pkg & db[, "Version"] == ver, ]
          }
        }
      )
    )
  ) > 0
}
```

or 

```r
check_if_pkg_available <- function(pkg, ver = NULL) {
  length(
    available.packages(
      filter = list(
        add = TRUE,
        function(db) {
          ver_flag <- if (is.null(ver)) TRUE else db[, "Version"] == ver
          db[db[, "Package"] == pkg & ver_flag, ]
        }
      )
    )
  ) > 0
}
```

Reply:
thanks - applied the first one